coordinator/store/store.go

```diff
@@ -16,6 +16,8 @@ type Store interface {
 	Get(string) ([]byte, error)
 	// Put saves a value to store by key
 	Put(string, []byte) error
+	// Iterator returns a list of keys with a given prefix
+	Iterator(string) ([]string, error)
```

Comment:
Not sure what's the idiomatic golang way here.
Iterator could be an interface that has `next` and `hasNext` functions.
Using a list we need to assume we can hold all data of the store in memory.
Should be ok in most cases.
Got an opinion @thomasten?

Reply:
Please use a `next`, `hasNext` interface.